### ISSUE TITLE
Non-deterministic compiled_safe_decomposed_exp and ExpPolicy simplification

### ISSUE THREAD

**Issue by Concode0** [Concode0/Versor#14]
### TL;DR

compiled_safe_decomposed_exp is non-deterministic (RNG-init power iteration), so two algebra.exp(B) calls on the same B return different rotors — silently breaks inverse(forward(x)) == x for n≥4. Separately, the current ExpPolicy axis (FAST/AUTO/EXACT) is misaligned with the real cost lever (iteration count); proposing reduction to BALANCED/PRECISE.

### Affected Area

core/decomposition.py

### Algebra Signature

where n>=4 ( that engages _exp_compiled_safe )

### What's Happening

```shell
# experiments/inc_lattice_morph.py --dim 4 (float64, default policy=AUTO)
# After 50 training steps:
Reconstruction Verification (tol=1e-05):
      b_1 error: 2.95e-05
      b_2 error: 1.82e-03
      b_3 error: 2.73e-05
      b_4 error: 6.76e-05
      Max error: 1.82e-03 -> FAIL

Root cause: _power_iteration_compiled_safe (core/decomposition.py:243-286) seeds its iteration vector from torch.randn per call:

  v_raw = torch.randn(*batch_shape, algebra.n, device=device, dtype=dtype)

Two consecutive algebra.exp(B) on the same B start from different v, converge to different (within-tolerance) approximations, and return different rotors. MorphStage.inverse() recomputed R = exp(B) independently from forward, so the inverse rotor was no longer the inverse of the forward rotor. Errors compound across stages.

Related design issue — ExpPolicy taxonomy:

ExpPolicy will be reconstructed with PRECISE, BALANCED ( FAST is deprecated cause too high errors over tolerance. )
```

### Expected Behavior

1. algebra.exp(B) should be a deterministic function of B and the algebra — calling it twice returns bit-identical tensors. 
inverse(forward(x)) == x to machine eps without caller-side rotor caching.

2. Policy enum should exposes the real lever (iteration budget vs accuracy target), not the implementation branch:

- BALANCED (default) — dtype/n-tuned iteration budget at the cost-benefit knee (bf16=16, f32=32, f64=96, optionally scale with n).
- PRECISE — iteration budget sized to saturate at machine eps (e.g., bf16=16, f32=128, f64=384).
- FAST will be removed.
- Branch dispatch (closed-form ↔ decomposition) becomes an internal detail driven by the simplicity test.

### Minimal Reproduction

```python
uv run python -m experiments.inc_lattice_morph --dim 4 --steps 50 --dtype float64
```

### Checks Already Tried (for instability / performance)

- [x] Reduced learning rate
- [x] Reduced bivector norm clip (`optimizer.bv_clip`)
- [ ] Switched to standard Adam
- [ ] Reduced model depth / channels
- [ ] Used `.reshape()` instead of `.view()` after multi-rotor ops
- [x] Passed `device='cpu'` explicitly to `CliffordAlgebra`
- [x] Tried a different `exp_policy` (`auto` / `fast` / `exact`)

### Device

- [ ] CPU
- [ ] CUDA
- [ ] Apple Silicon (MPS)

### Versor Commit / Python / PyTorch

5c86f13, Python 3.11.12, PyTorch 2.10.0
